worlds/rogue_legacy/__init__.py

```diff
@@ -49,6 +49,30 @@ def fill_slot_data(self) -> dict:
         return {option_name: self.get_setting(option_name).value for option_name in rl_options}
 
     def generate_early(self):
+        location_ids_used_per_game = {
+            world.game: set(world.location_id_to_name) for world in self.multiworld.worlds.values()
+        }
+        item_ids_used_per_game = {
+            world.game: set(world.item_id_to_name) for world in self.multiworld.worlds.values()
+        }
+        overlapping_games = set()
+
+        for id_lookup in (location_ids_used_per_game, item_ids_used_per_game):
+            for game_1, ids_1 in id_lookup.items():
+                for game_2, ids_2 in id_lookup.items():
+                    if game_1 == game_2:
+                        continue
+
+                    if ids_1 & ids_2:
+                        overlapping_games.add(tuple(sorted([game_1, game_2])))
+
+        if overlapping_games:
+            raise RuntimeError(
```

Comment:
Could we also maybe list said IDs? Or, if we expect the number to be very high, maybe talk about the first instance, and say the number of instances?

In the case of pokemon crystal, I imagine it's like, the entire ID set, but in some cases it might be just one item and it might be nice, especially for unsupported maintainers, to be told, as it is sometimes a bug